packages/@expo/server/src/index.ts

```diff
@@ -79,55 +78,47 @@ export function createRequestHandler(
       }
       return import(filePath);
     },
-    logApiRouteExecutionError = (error: Error) => {
-      console.error(error);
-    },
-    handleApiRouteError = async (error: Error) => {
-      if ('statusCode' in error && typeof error.statusCode === 'number') {
-        return new Response(error.message, {
-          status: error.statusCode,
-          headers: { 'Content-Type': 'text/plain' },
-        });
-      }
-      return new Response('Internal server error', {
-        status: 500,
-        headers: { 'Content-Type': 'text/plain' },
-      });
+    handleRouteError = async (error: Error) => {
+      // In production the server should handle unexpected errors
+      throw error;
     },
   }: {
-    getHtml?: (request: Request, route: RouteInfo<RegExp>) => Promise<string | Response | null>;
-    getRoutesManifest?: (distFolder: string) => Promise<ExpoRoutesManifestV1<RegExp> | null>;
-    getApiRoute?: (route: RouteInfo<RegExp>) => Promise<any>;
+    getHtml?: (request: Request, route: Route) => Promise<string | Response | null>;
+    getRoutesManifest?: (distFolder: string) => Promise<Manifest | null>;
+    getApiRoute?: (route: Route) => Promise<any>;
     logApiRouteExecutionError?: (error: Error) => void;
-    handleApiRouteError?: (error: Error) => Promise<Response>;
+    handleRouteError?: (error: Error) => Promise<Response>;
   } = {}
 ) {
-  let routesManifest: ExpoRoutesManifestV1<RegExp> | undefined;
+  let routesManifest: Manifest | undefined;
 
-  return async function handler(request: Request): Promise<Response> {
+  const getRoutesManifestCached = async () => {
+    let manifest: Manifest | null = null;
     if (getInternalRoutesManifest) {
-      const manifest = await getInternalRoutesManifest(distFolder);
-      if (manifest) {
-        routesManifest = manifest;
-      } else {
-        // Development error when Expo Router is not setup.
-        return new Response('No routes manifest found', {
-          status: 404,
-          headers: { 'Content-Type': 'text/plain' },
-        });
-      }
+      // Only used for development by the dev server
+      manifest = await getInternalRoutesManifest(distFolder);
     } else if (!routesManifest) {
-      routesManifest = getRoutesManifest(distFolder);
+      // Production
+      manifest = await getRoutesManifest(distFolder);
     }
 
-    const url = new URL(request.url, 'http://expo.dev');
+    if (manifest) {
+      routesManifest = manifest;
+    }
+
+    return routesManifest;
+  };
+
+  async function requestHandler(incomingRequest: Request, manifest: Manifest) {
+    let request = incomingRequest;
+    const url = new URL(request.url);
 
     let sanitizedPathname = url.pathname;
 
     debug('Request', sanitizedPathname);
```

Comment:
future task: Probably easier to just remove/move this in some way with the handlers. We can add a helper for that. I suspect once we start splitting this into universal & non-universal parts it'll become clearer, but generally, this isn't helpful to have here imho